### DISCUSSION TITLE
timeout

### DISCUSSION THREAD

**Discussion by llingy** [marp-team/marp#225] [Q&A] [answered]
`[ ERROR ] Failed converting Markdown. (Navigation timeout of 30000 ms exceeded)`
What should I do

**Comment by yhatt** [comment]
- Confirm if you have a stable connection to network. Marp conversion may require to fetch network resources while conversion, such as contents used in Markdown, Web Fonts by the theme CSS, and KaTeX Web fonts provided by [jsDelivr CDN](https://www.jsdelivr.com/).
- Try HTML export instead of PDF and PPTX. It's free of any timeouts. And you also can try to convert HTML into PDF by using Chrome/Chromium-flavored browser (Menu -> "Print" -> Choose "Save as PDF" in the Destination)

**Reply by llingy** [reply]
**Thank you**

**Comment by yhatt** [comment]
**UPDATE:** Marp CLI v1.5.1 and later can set custom time-out period for internal Puppeteer actions by `PUPPETEER_TIMEOUT` env in milliseconds. (https://github.com/marp-team/marp-cli/pull/409)

```
PUPPETEER_TIMEOUT=60000 marp --pdf large-markdown.md
```

If you want to disable timeout, set `PUPPETEER_TIMEOUT=0`.

**Reply by Yan1025** [reply]
> **UPDATE:** Marp CLI v1.5.1 and later can set custom time-out period for internal Puppeteer actions by `PUPPETEER_TIMEOUT` env in milliseconds. ([marp-team/marp-cli#409](https://github.com/marp-team/marp-cli/pull/409))
> 
> ```
> PUPPETEER_TIMEOUT=60000 marp --pdf large-markdown.md
> ```
> 
> If you want to disable timeout, set `PUPPETEER_TIMEOUT=0`.

Hi, is there any solution to config PUPPETEER_TIMEOUT in vscode plugin of  marp?

**Reply by yhatt** [reply]
Marp for VS Code is using an internally bundled Marp CLI for the export command, so `PUPPETEER_TIMEOUT` setting is still working also in Marp for VS Code extension if VS Code itself has recognized that environment value.

Try [launching VS Code from CLI](https://code.visualstudio.com/docs/editor/command-line#_launching-from-command-line) with session environment variable setting such as following:

```bash
# macOS/Linux
PUPPETEER_TIMEOUT=XXX code
```

```cmd
# Windows
set PUPPETEER_TIMEOUT=XXX
code
```

You also can set `PUPPETEER_TIMEOUT` to environment variables for either of system and user, to apply setting permanently.

> **Note** You may follow extra steps to check whether VS Code is detecting environment variable you've set:
>
> 1. Select [Help] menu → [Toggle Developer Tools]
> 2. Type in `vscode.process.env.PUPPETEER_TIMEOUT` to [Console] tab in the developer tools
> 3. Hit <kbd>Enter</kbd>, and check the result whether reflected the custom variable value you've set.

**Reply by Yan1025** [reply]
> Marp for VS Code is using an internally bundled Marp CLI for the export command, so `PUPPETEER_TIMEOUT` setting is still working also in Marp for VS Code extension if VS Code itself has recognized that environment value.
> 
> Try [launching VS Code from CLI](https://code.visualstudio.com/docs/editor/command-line#_launching-from-command-line) with session environment variable setting such as following:
> 
> ```shell
> # macOS/Linux
> PUPPETEER_TIMEOUT=XXX code
> ```
> 
> ```batchfile
> # Windows
> set PUPPETEER_TIMEOUT=XXX
> code
> ```
> 
> You also can set `PUPPETEER_TIMEOUT` to environment variables for either of system and user, to apply setting permanently.
> 
> > Note You may follow extra steps to check whether VS Code is detecting environment variable you've set:
> > 
> > 1. Select [Help] menu → [Toggle Developer Tools]
> > 2. Type in `vscode.process.env.PUPPETEER_TIMEOUT` to [Console] tab in the developer tools
> > 3. Hit Enter, and check the result whether reflected the custom variable value you've set.

It works. Thanks a lot for your detailed answer

**Reply by vmgustavo** [reply]
I am having the same problem. I don't know if it is only because the presentation is big (33 pages) or if there is something else breaking it, but I am using `PUPPETEER_TIMEOUT=600000` and it still didn't finish before timeout. Is there anything I could do to help debug this?

**Reply by yhatt** [reply]
Try `--debug=true` or `--debug=all` option: https://github.com/marp-team/marp-cli/issues/646#issuecomment-2643389636

**Reply by vmgustavo** [reply]
Sorry I have not been clear enough. I am using VSCode, is there an option to debug through it? Something similar to the pupeteer env var maybe

**Reply by yhatt** [reply]
Marp for VS Code outputs logs into the console, that can show in the developer tools (`Developer: Toggle Developer Tools`), but it does not provide an extension setting to show detailed debug logs.

If you want it, try to launch VS Code with `DEBUG` environment value as `marp-cli*` (Output logs with the filter of `marp-cli` prefix: equivalent to `--debug=true`) or `*` (Logs from all dependencies that includes Puppeteer: equivalent to `--debug=all`).  It is backed by [`debug` module](https://github.com/debug-js/debug).

```
DEBUG="marp-cli*"
```

```
DEBUG="*"
```



Related: https://github.com/marp-team/marp-cli/blob/main/.github/CONTRIBUTING.md#debug

**Reply by vmgustavo** [reply]
Just for whoever might get into a similar problem in the future: the main problem in my case was not having chrome installed, it was using firefox by default and timing out. After installing chrome it was an instant execution.

**Comment by yhatt** [comment, marked as answer]
**UPDATE**: [Marp CLI v4](https://github.com/orgs/marp-team/discussions/542) and later versions are providing `--browser-timeout` option to set the custom timeout duration for the browser that is used for generating PDF, image(s), and PowerPoint document.

```shell
# Set the browser timeout to 60 seconds
marp foobar.md --pdf --browser-timeout 60

# Disable the browser timeout by setting it to 0 seconds
marp foobar.md --pdf --browser-timeout 0
```
